src/device/pf_cmina.c

```diff
@@ -829,3 +853,150 @@ void pf_cmina_show(
           p_cfg->eth_addr.addr[4],
           p_cfg->eth_addr.addr[5]);
 }
+
+/************************* Validate incoming data ***************************/
+
+/**
+ * @internal
+ * Check if a station name is valid
+ *
+ * Defined in Profinet 2.4, section 4.3.1.4.16.2:
+ *
+ * - 1 or more labels, separated by [.]
+ * - Total length is 1 to 240
+ * - Label length is 1 to 63
+ * - Labels consist of [a-z0-9-]
+ * - Labels do not start with [-]
+ * - Labels do not end with [-]
+ * - Labels do not use multiple concatenated [-] except for IETF RFC 5890
+ * - The first label does not have the form “port-xyz” or “port-xyz-abcde”
+ *     with a, b, c, d, e, x, y, z = 0...9
+ * - Station-names do not have the form a.b.c.d with a, b, c, d = 0...999
+ *
+ * Also empty station name is valid here (indicating that no station name has been set).
+ *
+ * @param station_name     In: Station name
+ * @param len              In: Length of incoming string without terminator.
+ * @return  0  if the name is valid
+ *          -1 if the name is invalid
+ */
+int pf_cmina_is_stationname_valid(
```

Comment:
These validation functions effectively compute boolean values and should probably return bool?

Reply:
OK, changed return type to bool